### ISSUE TITLE
Resolving Trailing Slash Issue in Breadcrumbs Links

### ISSUE THREAD

**Issue by shah-iq** [felix-berlin/astro-breadcrumbs#212]
I'm encountering an issue with the using astro-breadcrumbs with Astrowind template on my website. I'm using the astro-breadcrumbs integration to display breadcrumbs. However, after running the build command `pnpm run build`, the links in the breadcrumbs are having trailing slashes appended to them, despite having set the trailing slash option to "never." Additionally, I've included `trailingSlash={false}` in the Breadcrumb component for added assurance. It seems the problem only pertains to the links generated by the breadcrumbs. Can you suggest any potential solutions?

**Comment by felix-berlin** [comment]
Hi @shah-iq thanks for creating this issue. Do you have a link to your repo?

**Comment by shah-iq** [comment]
> Hi @shah-iq thanks for creating this issue. Do you have a link to your repo?

Actually this is a project for a client and is a private repo

**Comment by felix-berlin** [comment]
@shah-iq I will check this issue within this (hopefully) week. By the way the traillingSlash prop has been deprecated since V2. Please make sure you read the Migration Guide https://docs.astro-breadcrumbs.kasimir.dev/guides/migration-to-v2/

**Comment by felix-berlin** [comment]
Did you use the current release?

**Comment by shah-iq** [comment]
> Did you use the current release?

Yes I am using the latest release and I used the `traillingSlash` prop only to just double check if by any chance resolved the issue. I had seen the migration guide that this prop had been deprecated.

**Comment by felix-berlin** [comment]
@shah-iq is the problem still there? Otherwise I would close the issue.

**Comment by ccoyotedev** [comment]
Hey @felix-berlin 

My team is also encountering this issue. It work's fine in dev mode, but in the production build the links are appending a `/` even if there isn't one in the URL.

This is problematic for SEO

**Comment by felix-berlin** [comment]
Hi @ccoyotedev ,

can you maybe provide a working demo of your problem?

When this is not possible, can you tell me your current astro-breadcrumb version, astro version, used adapters, astro config and the astro-breadcrumbs usage?

**Comment by ccoyotedev** [comment]
@felix-berlin I cannot share as it's a private repository for the company I work for.

```json
"astro": "^3.6.5",
"astro-breadcrumbs": "^3.0.1",
```

In production it is a static build, hosted with Cloudflare.
```js
// astro.config.mjs

export default defineConfig({
  output: process.env.PUBLIC_APP_ENV === 'preview' ? 'server' : 'static',
  adapter: process.env.PUBLIC_APP_ENV === 'preview' ? node({ mode: 'standalone' }) : undefined,
  site: SITE_URL,
  trailingSlash: process.env.PUBLIC_APP_ENV === 'preview' ? 'ignore' : 'never',
  integrations: [
    react(),
    tailwind({
      config: {
        applyBaseStyles: false
      }
    }),
    // https://docs.astro.build/en/guides/integrations-guide/sitemap/
    sitemap({
      customPages: externalPages
    }),
    storyblok({
      ...
    })
  ]
})
```

Used like so:
```jsx
<Breadcrumbs linkTextFormat="capitalized">
  <span slot="separator">/</span>
</Breadcrumbs>
```

**Comment by felix-berlin** [comment]
I just tried to reproduce this behavior with the docs site, there I also use Cloudflare (Pages). But had no luck.

![image](https://github.com/user-attachments/assets/2d2d7d43-8558-4926-91c8-f35a70b1772c)

I checked whether `hasTrailingSlash` shows any abnormalities. This was not the case, at this point I am now a little perplexed.

If `const hasTrailingSlash = Astro.url.pathname.endsWith("/");` returns a wrong result under certain circumstances, I could try to query this logic without the values on `Astro.url`, i.e. do the whole thing directly in `generateCrumbs()`.
